5 Arrays and Cycles.js

```diff
@@ -1,99 +1,97 @@
 console.log(str);
-arr = str.split(',');
-console.log(arr);
-
+manufactures = str.split(",");
+console.log(manufactures);
 
 // Преобразовать числовой массив в Boolean
-let arr = [1,0,7,8,0,0,5,6];
-let result = arr.map(function(item, index, array){
-    return Boolean(item);
+let numbersToBoolean = [1, 0, 7, 8, 0, 0, 5, 6];
+let boolArray = numbersToBoolean.map(function (item, index, array) {
+  return Boolean(item);
 });
-console.log(result);
+console.log(boolArray);
 
 //Отфильтровать массив [1,6,7,8,3,4,5,6] по значению >3
 
-let arr = [1,6,7,8,3,4,5,6];
-let result = []
-for (let i=0; i<arr.length; i++){
-    if (arr[i]>3){
-        result.push(arr[i]);
-    };
-};
-console.log(result);
-
+let numbersToFilter = [1, 6, 7, 8, 3, 4, 5, 6];
+let filtered = numbersToFilter.filter((n) => n > 3);
+console.log(filtered);
 
 //Отсортировать массив [1,6,7,8,3,4,5,6] по убыванию
-let arr = [1,6,7,8,3,4,5,6];
-arr.sort(function(a,b){
-    return b - a;
+let decreaseNumbers = [1, 6, 7, 8, 3, 4, 5, 6];
+decreaseNumbers.sort(function (a, b) {
+  return b - a;
 });
-console.log(arr)
-
+console.log(decreaseNumbers);
 
 //Написать функцию, которая принимает два параметра - массив и число и выводит индекс элемента массива равный числу
 
-function callArray(arr, n){
-    for(let i = 0; i < arr.length; i++){
-        if (arr[i]==n){
-            return i;
-            break;
-        };
-    };
-};
-console.log(callArray([8, 7, 6, 6, 5, 4, 3, 1], 6))
-
-
+function callArray(indexArray, n) {
+  for (let i = 0; i < indexArray.length; i++) {
+    if (indexArray[i] === n) {
+      return i;
+    }
+  }
+}
+console.log(callArray([8, 7, 6, 6, 5, 4, 3, 1], 6));
 
 //Реализовать цикл, который будет выводить число а, пока оно не станет меньше 10
 
-let a = prompt('Enter a');
-while (a < 10) {
+function lessThan10(a) {
+  while (a >= 10) {
     console.log(a);
-    a = prompt("Enter next value");
-  };
-
+    a--;
+  }
+}
 
 //Реализовать цикл, который выводит в консоль простые числа
 
-console.log(1);
-let n = 3;
-nextPrime:
-for (let i = 2; i <= n; i++) {
-  for (let j = 2; j < i; j++) { 
-    if (i % j == 0) continue nextPrime; 
-  };
-  console.log(i);
-  i = Number(prompt("press enter for the next prime", i)); 
-  if (i == 1) break;
-  n=n+i;
-};
-
+function isPrimeNumber(x) {
+  if (x === 2) return x;
+  for (let n = 2; n < x; n++) {
+    if (x % n === 0) {
+      return false;
+    }
+  }
+  return x;
+}
+
+function primeNumberList(nmbr) {
+  for (let k = 2; k <= nmbr; k++) {
+    if (isPrimeNumber(k)) {
+      console.log(k);
+    }
+  }
+}
 
 //Реализовать цикл, который выводит в консоль нечетные числа
 
-let a = 1;
-while (a){
-    console.log(a);
-    a = Number(prompt('press enter for next number', a+=2));
-}
\ No newline at end of file
+function oddNumbers(num) {
+  if (num > 0) {
+    for (let j = 1; j <= num; j += 2) {
+      console.log(j);
+    }
+  } else {
+    for (let j = -1; j >= num; j -= 2) {
+      console.log(j);
+    }
+  }
+}
```

Comment:
It works but it is not a good solution. I'll show an example:
```
function printOdds(from, to) {
  console.log('Odd numbers:');
  for (let i = from; i <= to; i++) {
    if (i % 2 !== 0) {
      console.log(i);
    }
  }
}

printOdds(-15, 22);
```

Reply:
In this case we don't need to bother which numbers we are checking: positive or negative.